python/ray/rllib/agents/pg/pg.py

```diff
@@ -22,40 +21,24 @@
 # yapf: enable
 
 
-class PGTrainer(Trainer):
-    """Simple policy gradient agent.
-
-    This is an example agent to show how to implement algorithms in RLlib.
-    In most cases, you will probably want to use the PPO agent instead.
-    """
-
-    _name = "PG"
-    _default_config = DEFAULT_CONFIG
-    _policy_graph = PGPolicyGraph
-
-    @override(Trainer)
-    def _init(self, config, env_creator):
-        if config["use_pytorch"]:
-            from ray.rllib.agents.pg.torch_pg_policy_graph import \
-                PGTorchPolicyGraph
-            policy_cls = PGTorchPolicyGraph
-        else:
-            policy_cls = self._policy_graph
-        self.local_evaluator = self.make_local_evaluator(
-            env_creator, policy_cls)
-        self.remote_evaluators = self.make_remote_evaluators(
-            env_creator, policy_cls, config["num_workers"])
-        optimizer_config = dict(
-            config["optimizer"],
-            **{"train_batch_size": config["train_batch_size"]})
-        self.optimizer = SyncSamplesOptimizer(
-            self.local_evaluator, self.remote_evaluators, **optimizer_config)
-
-    @override(Trainer)
-    def _train(self):
-        prev_steps = self.optimizer.num_steps_sampled
-        self.optimizer.step()
-        result = self.collect_metrics()
-        result.update(timesteps_this_iter=self.optimizer.num_steps_sampled -
-                      prev_steps)
-        return result
+def _make_policy_optimizer(local_ev, remote_evs, config):
+    optimizer_config = dict(config["optimizer"],
+                            **{"train_batch_size": config["train_batch_size"]})
+    return SyncSamplesOptimizer(local_ev, remote_evs, **optimizer_config)
+
+
+def _get_policy_graph(config):
+    if config["use_pytorch"]:
+        from ray.rllib.agents.pg.torch_pg_policy_graph import \
+            PGTorchPolicyGraph
+        return PGTorchPolicyGraph
+    else:
+        return PGPolicyGraph
+
+
+PGTrainer = build_trainer(
+    "PG",
+    default_config=DEFAULT_CONFIG,
+    default_policy_graph=PGPolicyGraph,
```

Comment:
why do you still need DEFAULT Policy graph if you already have the creator below?

Reply:
Just for consistency with existing algorithms that assign _policy_graph. Also, it is optional.